### ISSUE TITLE
Boolean/set operations

### ISSUE THREAD

**Issue by n8willis** [simoncozens/beziers.py#15]
Any thoughts on adding union/remove-overlap functionality for Path objects? 

(Other set-manipulation functions might also be useful, certainly; this is just the one that is most immediately applicable for the sorts of things I'm doing, and my fontTools searching seems to suggest that overlap removal is a fairly commonly-felt need.)

**Comment by simoncozens** [comment]
`.union`, `.intersection` and `.difference` should work already.

**Comment by n8willis** [comment]
So, they're just not documented?

**Comment by simoncozens** [comment]
Yeah, sorry. That's also a bug. I thought I was being clever putting functionality into mixin modules (which is not very Pythonic, but is *much* neater than all-methods-in-the-same-file) but having stuff out of the main file makes it liable to get overlooked in documentation. Same goes for `SampleMixin`.

**Comment by n8willis** [comment]
Yeah; I at least don't get the mixin approach — or, at least, it does not do what I expect it to here. So I'm not clear if that's a docs issue or if the architecture doesn't match the functionality I perceive it to.

E.g., if you create a BezierPath from a fontTools glyph, then I would expect to be able to call the union function on the resultant BezierPath objects as a set (or list, or however you'd iterate through them). Instead it seems like it has to be called on a single path and given another single path? So, if I follow that correctly, then it's not really a set operation like I was expecting, but something piecewise instead. Then again, I'm really not sure I get the point or benefits of mixins, so maybe it's all fine?

**Comment by simoncozens** [comment]
You're right; I thought of the boolean shape operations as binary operations (a subtract b), but maybe it makes sense to see them as n-ary, in which case they should probably be class methods.

In general the mixin approach helps me to organise the code into logical units instead of trying to implement everything n one big class file, and to share code like `.sample` between different classes. Maybe I should import those methods into the class instead, though.

**Comment by n8willis** [comment]
> `.union`, `.intersection` and `.difference` should work already.

Again, I might be butting up against things that, if there were documentation, would be 100% clear, but calling .union on a path with itself seems to explode, when that ought to be a simple identity.

Another wrinkle here is that the `union/difference/intersection()` functions as they are currently defined accept only individual paths as arguments but, in turn, they return only lists. So it's non-trivial to just loop through the path-list of a glyph and union them all as if it were a set operation.
